### PR TITLE
en: add doc for canary upgrade tidb operator

### PR DESCRIPTION
**Pull request by ran-huang**
Signed-off-by: Ran <huangran@pingcap.com>

### What is changed, added, or deleted? (Required)

Add doc for canary upgrade tidb operator.

**FOLLOW-UP:**
> I'll remove v1.1-version-specific description from master later.

### Which TiDB Operator version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v1.1 (TiDB Operator 1.1 versions)
- [ ] v1.0 (TiDB Operator 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-tidb-operator/pull/1123
- Other reference link(s): 

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Comment by ran-huang** [comment]
@TomShawn PTAL

File: en/canary-upgrade-tidb-operator.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
+---
+title: Perform a Canary Upgrade on TiDB Operator
+summary: Learn how to perform a canary upgrade on TiDB Operator in Kubernetes.
+---
+
+# Perform a Canary Upgrade on TiDB Operator
+
+This document describes how to perform a canary upgrade on TiDB Operator. By canary upgrades, you can prevent TiDB Operator upgrade from causing unexpected impact on all the TiDB clusters in Kubernetes. After you confirm the impact of TiDB Operator upgrade or the upgraded TiDB Operator works stably, you can normally upgrade TiDB Operator.
```
Comment (TomShawn) [inline en/canary-upgrade-tidb-operator.md:8]:
```suggestion
This document describes how to perform a canary upgrade on TiDB Operator. Using canary upgrades, you can prevent normal TiDB Operator upgrade from causing an unexpected impact on all the TiDB clusters in Kubernetes. After you confirm the impact of TiDB Operator upgrade or that the upgraded TiDB Operator works stably, you can normally upgrade TiDB Operator.
```

File: en/canary-upgrade-tidb-operator.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+    Refer to [Upgrade TiDB Operator](upgrade-tidb-operator.md). Add the following configuration in the `values.yaml` file, and upgrade TiDB Operator:
+
+    ```yaml
+    controllerManager:
+      selector:
+      - version!=canary
+    ```
+
+    If you already performed the step above, skip to Step 2.
```
Comment (TomShawn) [inline en/canary-upgrade-tidb-operator.md:31]:
```suggestion
    If you have already performed the step above, skip to Step 2.
```

File: en/canary-upgrade-tidb-operator.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    ```
+
+    > **Note:**
+    >
+    > * It is recommended to deploy the new TiDB Operator in a separate namespace.
+    > * Set `appendReleaseSuffix` to `true`.
+    > * If you do not need to perform a canary upgrade on `tidb-scheduler`, configure `scheduler.create: false`.
+    > * If you configure `scheduler.create: true`, a scheduler named `{{ .scheduler.schedulerName }}-{{.Release.Name}}` will be created. To use this scheduler, configure `spec.schedulerName` in the `TidbCluster` CR to the name of this scheduler.
+
+3. To test the canary upgrade of `tidb-controller-manager`, set labels for some TiDB cluster by running the following command:
```
Comment (TomShawn) [inline en/canary-upgrade-tidb-operator.md:53]:
```suggestion
3. To test the canary upgrade of `tidb-controller-manager`, set labels for a TiDB cluster by running the following command:
```

**Review by TomShawn — APPROVED** [review]
rest
/lgtm

**Pull request merged**